core/modules/nat.cc

```diff
@@ -234,13 +237,15 @@ void NAT::ProcessBatch(bess::PacketBatch *batch) {
     }
 
     // Garbage collect
-    if (available_ports_.empty()) {
+    if (available_ports_.empty() && now >= next_expiry_) {
       for (auto &record : flow_vec_) {
         if (record.time != 0 && now - record.time >= TIME_OUT) {
           available_ports_.push_back(record.port);
           record.time = 0;
           flow_hash_.erase(record.internal_flow);
           flow_hash_.erase(record.external_flow.ReverseFlow());
+        } else if (record.time != 0) {
+          next_expiry_ = std::min(next_expiry_, record.time + TIME_OUT);
```

Comment:
If I'm reading this right, won't the old next_expiry_ be the min always?  Do we need to set it to int max at the beginning of this if?